src/easylink/pipeline_schema.py

```diff
@@ -25,28 +57,79 @@ def __repr__(self) -> str:
     def validate_step(
         self, pipeline_config: LayeredConfigTree, input_data_config: LayeredConfigTree
     ) -> dict[str, list[str]]:
-        """Validates the pipeline configuration against this instance of pipeline schema.
+        """Validates the pipeline configuration against this ``PipelineSchema``.
+
+        Parameters
+        ----------
+        pipeline_config
+            The pipeline configuration to validate.
+        input_data_config
+            The input data configuration.
+
+        Returns
+        -------
+            A dictionary of errors, where the keys are the names of any steps that
+            did not validate and the values are lists of as many error messages as
+            could be generated for each of those steps.
 
         Notes
         -----
         Below, we nest the full pipeline configuration under a "substeps" key of
-        a root hierarchical step because the root step doesn't exist from the
-        user's perspective and it doesn't appear explicitly in the pipeline.yaml.
+        a root :class:`~easylink.step.HierarchicalStep` because such a root step
+        doesn't exist from the user's perspective and doesn't appear explicitly in
+        the user-provided pipeline specification file.
```

Comment:
It kind of does now, right? Can we simplify this by defining that a PipelineSchema's substeps key is called `steps`?

Reply:
Not sure I understand. I think it's still correct to say that the PipelineSchema is abstracted away from the user.

Reply:
Based on zoom discussion we agreed to leave this for now. It is admittedly clunky though